src/babel/index.js

```diff
@@ -700,6 +712,93 @@ function transformZacsAttributesOnNonZacsElement(t, platform, path) {
     .concat(addedAttrs)
 }
 
+function isPlainObjectProperty(t, node) {
+  return (
+    t.isObjectProperty(node) &&
+    t.isIdentifier(node.key) &&
+    !node.shorthand &&
+    !node.computed &&
+    !node.method
+  )
+}
+
+function validateStyleSheet(t, args, path) {
+  const [stylesheet] = args
+  if (!(args.length === 1 && t.isObjectExpression(stylesheet))) {
+    throw path.buildCodeFrameError('ZACS StyleSheet accepts a single Object argument')
+  }
+  const stylesets = stylesheet.properties
+  if (!stylesets.every(styleset => isPlainObjectProperty(t, styleset))) {
+    // TODO: We can probably allow `"name":`, no problem.
+    throw path.buildCodeFrameError(
+      'ZACS StyleSheet stylesets must be defined as `name: {}`. Other syntaxes, like `[name]:`, `"name": `, `...styles` are not allowed',
+    )
+  }
+
+  if (
+    !stylesets.every(
+      styleset =>
+        t.isObjectExpression(styleset.value) &&
+        styleset.value.properties.every(property => isPlainObjectProperty(t, property)),
+    )
+  ) {
+    throw path.buildCodeFrameError(
+      'ZACS StyleSheets can only contain simple style properties with properties like so: `{ backgroundColor: \'red\', height: 100 }`. Other syntaxes, like `[prop]:`, `"prop": `, `...styles` are not allowed.',
+    )
+  }
+
+  // TODO: Validate values
+}
+
+// TODO: Find a JS-to-CSS package
+// Inspiration: https://github.com/giuseppeg/style-sheet/blob/master/src/data.js
+const capitalRegex = /([A-Z])/g
+const cssCaseReplacer = (match, letter) => `-${letter.toLowerCase()}`
```

Comment:
Do you have a plan for handling things like pseudoselectors?

Reply:
yes! Coming in part 3